src/SA/ldapsecuritycheck/entry.c

```diff
@@ -0,0 +1,289 @@
+#define SECURITY_WIN32
+#include <windows.h>
+#include <security.h>
+#include "beacon.h"
+#include <winldap.h>
+#include <dsgetdc.h>
+#include "../../common/bofdefs.h"
+
+VERIFYSERVERCERT ServerCertCallback;
+BOOLEAN _cdecl ServerCertCallback (PLDAP Connection, PCCERT_CONTEXT pServerCert)
+{
+	return TRUE;
+}
+
+BOOL checkLDAP(wchar_t* dc, wchar_t* spn, BOOL ssl)
+{
+	CredHandle hCredential;
+	TimeStamp tsExpiry;
+	SECURITY_STATUS getHandle = SECUR32$AcquireCredentialsHandleW(
+		NULL,
+		L"NTLM",
+		SECPKG_CRED_OUTBOUND,
+		NULL,
+		NULL,
+		NULL,
+		NULL,
+		&hCredential,
+		&tsExpiry
+	);
+
+	if (getHandle != SEC_E_OK)
+	{
+		BeaconPrintf(CALLBACK_ERROR, "[-] AcquireCredentialsHandleW failed: %lu\n", getHandle);
+		return FALSE;
+	}
+
+	//ldap
+	ULONG result;
+	LDAP* pLdapConnection = NULL;
+
+	if(ssl == TRUE){
+		pLdapConnection = WLDAP32$ldap_initW(dc, 636);
+	}else{
+		pLdapConnection = WLDAP32$ldap_initW(dc, 389);
+	}
+	if (pLdapConnection == NULL)
+	{
+		BeaconPrintf(CALLBACK_ERROR, "[-] Failed to establish LDAP connection");
+		return FALSE;
+	}
+
+	const int version = LDAP_VERSION3;
+	result = WLDAP32$ldap_set_optionW(pLdapConnection, LDAP_OPT_VERSION, (void*)&version);
+
+	if(ssl == TRUE){
+        WLDAP32$ldap_get_optionW(pLdapConnection, LDAP_OPT_SSL, &result);  //LDAP_OPT_SSL
+        if (result == 0)
+            WLDAP32$ldap_set_optionW(pLdapConnection, LDAP_OPT_SSL, LDAP_OPT_ON);
+
+        WLDAP32$ldap_get_optionW(pLdapConnection, LDAP_OPT_SIGN, &result);  //LDAP_OPT_SIGN
+        if (result == 0)
+            WLDAP32$ldap_set_optionW(pLdapConnection, LDAP_OPT_SIGN, LDAP_OPT_ON);
+
+        WLDAP32$ldap_get_optionW(pLdapConnection, LDAP_OPT_ENCRYPT, &result);  //LDAP_OPT_ENCRYPT
+        if (result == 0)
+            WLDAP32$ldap_set_optionW(pLdapConnection, LDAP_OPT_ENCRYPT, LDAP_OPT_ON);
+
+        WLDAP32$ldap_set_optionW(pLdapConnection, LDAP_OPT_SERVER_CERTIFICATE, (void*)&ServerCertCallback ); //LDAP_OPT_SERVER_CERTIFICATE
+	}
+
+	result = WLDAP32$ldap_connect(pLdapConnection, NULL);
+	if (result != LDAP_SUCCESS)
+	{
+		BeaconPrintf(CALLBACK_ERROR, "[-] ldap_connect failed:");
+		return FALSE;
+	}
+
+	ULONG res;
+	struct berval* servresp = NULL;
+
+	SecBufferDesc InBuffDesc;
+	SecBuffer InSecBuff;
+
+	SECURITY_STATUS initSecurity;
+	CtxtHandle newContext;
+
+	SecBuffer secbufPointer = { 0, SECBUFFER_TOKEN, NULL };
+	SecBufferDesc output = { SECBUFFER_VERSION, 1, &secbufPointer };
+
+	SecBuffer secbufPointer3 = { 0, SECBUFFER_TOKEN, NULL };
```

Comment:
This variable isn't used, with output2 and output sharing the same backing buffer of secbufPointer